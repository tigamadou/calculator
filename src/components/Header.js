import React from 'react';
import {
  Link,
} from 'react-router-dom';

const Header = () => (
  <header className="header">
    <div className="pageTitle">
      Math Magicians
    </div>
    <nav className="navbar">
      <ul className="navbarMenu">
        <li className="navbarItem">
          <Link to="/">Home</Link>
        </li>
        <li className="navbarItem">
          <Link to="/calculator">Calculator</Link>
        </li>
        <li className="navbarItem">
          <Link to="/quote">Quote</Link>
        </li>
      </ul>
    </nav>
  </header>
);

export default Header;
