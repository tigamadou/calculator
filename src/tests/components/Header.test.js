import React from 'react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../../components/Header';

describe('Header component tests', () => {
  test('Header contains Text', () => {
    render(<Header />, { wrapper: MemoryRouter });
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('Math Magicians')).toBeInTheDocument;
  });
  test('Header should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
