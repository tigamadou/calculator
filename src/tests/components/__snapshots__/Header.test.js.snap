// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests Header should match with snapshot 1`] = `
<header
  className="header"
>
  <div
    className="pageTitle"
  >
    Math Magicians
  </div>
  <nav
    className="navbar"
  >
    <ul
      className="navbarMenu"
    >
      <li
        className="navbarItem"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="navbarItem"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="navbarItem"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
